main.tf

```diff
@@ -25,17 +25,10 @@ resource "aws_iam_policy" "main" {
   path        = var.path
   description = var.policy_description
   policy      = var.policy
-
 }
 
 resource "aws_iam_role_policy_attachment" "main" {
   count      = var.policy != null ? 1 : 0
   role       = aws_iam_role.main.name
   policy_arn = join("", aws_iam_policy.main.*.arn)
 }
-
-resource "aws_iam_role_policy_attachment" "managed" {
-  count      = length(var.managed_policy_arns) > 0 ? length(var.managed_policy_arns) : 0
-  role       = aws_iam_role.main.name
```

Comment:
Wouldn't this be a good thing to maintain? you might want to attach these policies in the future.

Reply:
What the resource aimed to achieve was already being done by the `managed_policy_arns` argument. The aws-managed policies can be added directly to the list..